### PR TITLE
Refactoring of tests and error handling

### PR DESCRIPTION
**Pull request by neilbmclaughlin**
* Added some HTML parsing for more specific tests. This is to aid debugging of failing tests and to improve the specificity of the tests.
* Added individual 'sub' route handlers for the different river and sea level listings
* Move from returning a single location search result to an array with one element of search results with a nod towards the upcoming disambiguation work and error handling (see below)
* Move from throwing an error when no location results are found to returning an empty array.
* Move search error error handling up to the error plugin

Notes regarding tests
* not sure if the preference would be for id's or classes to be added to the ul's
* also not sure if adding messages to the expects adds any value
* dev dependencies seem to be added to dependencies

### CODE REVIEW

**Comment by neilbmclaughlin** [comment]
Just for info, the SonarCloud is reporting lots of code issues but this is a temporary thing until I can remove the original ViewModel once it's no longer used.

File: server/models/views/river-and-sea-levels.js [hunk 1]
```diff
@@ -1,287 +1,310 @@
...
+      stationProperties(station, place, stations, originalStation)
+    })
+    stations.sort((a, b) => a.distance - b.distance)
+
+    // ref: https://flaviocopes.com/javascript-destructure-object-to-existing-variable/
+    ;({ filters, activeFilter } = setFilters(stations, queryGroup))
+
+    deleteUndefinedProperties(stations)
+
+    stations = isEngland ? stations : []
```
Comment (nikiwycherley) [inline server/models/views/river-and-sea-levels.js:191]: ternary operator is inside a check that has already checked for stations

File: server/models/views/river-and-sea-levels.js [hunk 2]
```diff
@@ -1,287 +1,310 @@
...
+    queryGroup = activeFilter
+  }
+  stations = isEngland ? stations : []
```
Comment (nikiwycherley) [inline server/models/views/river-and-sea-levels.js:194]: same ternary operator as above, just duplication of code

File: server/models/views/river-and-sea-levels.js [hunk 3]
```diff
@@ -1,287 +1,310 @@
...
-      referer: referer,
-      rivers: this.getRiverNames(stations),
-      types: this.getTypes(stations),
-      taCode: targetArea && targetArea.fws_tacode,
-      isEngland: place ? place.isEngland.is_england : null,
-      riverId: this.getRiverId(riverIds),
-      stationsBbox: [],
-      originalStation: stations.originalStation
-    })
+const { bingKeyMaps, floodRiskUrl } = require('../../config')
```
Comment (nikiwycherley) [inline server/models/views/river-and-sea-levels.js:4]: functions are above view model function

File: server/models/views/river-and-sea-levels.js (hunk 1 above)
Comment (neilbmclaughlin) [reply server/models/views/river-and-sea-levels.js:191]: Removed in #348

File: server/models/views/river-and-sea-levels.js (hunk 2 above)
Comment (neilbmclaughlin) [reply server/models/views/river-and-sea-levels.js:194]: Removed in #345

File: server/models/views/river-and-sea-levels.js (hunk 3 above)
Comment (neilbmclaughlin) [reply server/models/views/river-and-sea-levels.js:4]: Moved in #345

**Comment by neilbmclaughlin** [comment]
Closing this PR. New branch and PR created following rebase of base branch.

**Pull request closed without merging**